### ISSUE TITLE
wcs2:post-xml-req5 fails with InvocationTargetException (post/xml)

### ISSUE THREAD

**Issue by lgoltz** [opengeospatial/ets-wcs20#55]
## Setup
SUT: TODO
Selected CCs: all, except _WCS WPS 2.0 (Processing)_
Tested locally with docker environment 
  * ETS WCS 2. 0 1.13-SNAPSHOT 
 * TEAM Engine 5.2-SNAPSHOT

## Testfailure
"Error in call to extension function {public org.w3c.dom.NodeList com.occamlab.te.TECore.request(org.w3c.dom.Document,java.lang.String) throws java.lang.Throwable}: Exception in extension function java.lang.reflect.InvocationTargetException"

* Service returns a multipart response. ETS fails with the following failure (error_log.txt):
```
Caused by: java.lang.RuntimeException: Failed to parse resource from https://ogc.webgistesting.net/server/services/Mynetcdf/ImageServer/WCSServer 
 Failed to parse input: sun.net.www.protocol.http.HttpURLConnection$HttpInputStream
        at com.occamlab.te.parsers.XMLValidatingParser.parse(XMLValidatingParser.java:222)
        ... 83 more
```

## Proposal
 * Further investigations required

**Comment by lgoltz** [comment]
The failure occurs cause the GetCoverage Response is tried to be parsed as XML. Instead the HTTPParser should be used as in GetCoverage.ctl.

**Comment by keshav-nangare** [comment]
@lgoltz 

Service is not available to reproduce this behavior. Could you please provide any other service which will reproduce the same behavior?

**Comment by keshav-nangare** [comment]
@lgoltz @dstenger 

I have tested the Zikang's service with this test and test gets failed with the same error.

Lyn, you are right the service response is multipart so the XMLValidatingparser is failed to parse the multipart data because it expecting the only XML.

```
Request d2814e456_1:
   Method: post
   URL: http://34.207.148.210:6080/arcgis/services/netcdf2/ImageServer/WCSServer
   Body: 
<wcs:GetCoverage xmlns:ctl="http://www.occamlab.com/ctl"
                 xmlns:ctlp="http://www.occamlab.com/te/parsers"
                 xmlns:gen="java:com.occamlab.te.Generator"
                 xmlns:gml="http://www.opengis.net/gml/3.2"
                 xmlns:ows="http://www.opengis.net/ows/2.0"
                 xmlns:saxon="http://saxon.sf.net/"
                 xmlns:sch="http://purl.oclc.org/dsdl/schematron"
                 xmlns:te="http://www.occamlab.com/te"
                 xmlns:tec="java:com.occamlab.te.TECore"
                 xmlns:tems="java:com.occamlab.te.web.MonitorServlet"
                 xmlns:wcs="http://www.opengis.net/wcs/2.0"
                 xmlns:wcs2="http://www.opengis.net/wcs/2.0"
                 xmlns:xlink="http://www.w3.org/1999/xlink"
                 xmlns:xs="http://www.w3.org/2001/XMLSchema"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                 service="WCS"
                 version="2.0.1"
                 xsi:schemaLocation="http://www.opengis.net/wcs/2.0 ../../wcsAll.xsd">
   <wcs:CoverageId>netcdf2</wcs:CoverageId>
</wcs:GetCoverage>
```

**Comment by dstenger** [comment]
I don't see a reason why multipart should be forbidden.

Thus, there must either a precondition be defined for test suite or multipart support must be enabled.

**Comment by lgoltz** [comment]
@keshav-nangare: Can you check if multipart support can be enabled in the test suite?

**Comment by ghobona** [comment]
@keshav-nangare 

If it's a .tif image then you could use Java's ImageIO class to check whether it is an image.

For example, the sample code below returns the width and height of an image if the file is an image. If the file is not an image, then an Exception is thrown.

```java
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.imageio.ImageIO;

public class NewClass {
    public static void main(String[] args) {
        try {
            BufferedImage bi = ImageIO.read(new File("/Users/ogccite/Downloads/myimage.tif"));
             System.out.println("Check "+bi.getHeight()+ " "+ bi.getWidth());            
        } catch (IOException ex) {
            Logger.getLogger(NewClass.class.getName()).log(Level.SEVERE, null, ex);
        }
    }    
}

```

**Comment by ghobona** [comment]
@keshav-nangare @ghobona discussed on 2020-07-15.

This issue is related to https://github.com/opengeospatial/ets-wcs11/issues/41 because of the multipart aspect.

They may have the same solution.

**Comment by keshav-nangare** [comment]
Discussed on 22-07-2020 with @dstenger and @ghobona 

Proposed solution:

Add validation for a response with following content-type:

`"image/*" OR "application/x-hdf5"`

If the response has above content-type then test should pass otherwise fail

**Comment by keshav-nangare** [comment]
Fixed with #98 PR.
